Library/Homebrew/test/caveats_spec.rb

```diff
@@ -26,4 +26,35 @@ def caveats
       expect(described_class.new(f)).not_to be_empty
     end
   end
+
+  describe "#caveats" do
+    context "when f.plist is not nil" do
+      it "prints plist startup information when f.plist_startup is not nil" do
+        allow(f).to receive(:plist).and_return(true)
+        allow(f).to receive(:plist_startup).and_return(true)
+        expect(subject.caveats).to include("To have launchd start #{f.full_name} now and restart at startup:\n  sudo brew services start #{f.full_name}\n")
+      end
+
+      it "prints plist login information when f.plist_startup is nil" do
+        allow(f).to receive(:plist).and_return(true)
+        expect(subject.caveats).to include("To have launchd start #{f.full_name} now and restart at login:\n  brew services start #{f.full_name}\n")
+      end
+    end
+
+    context "when f.keg_only is not nil" do
+      it "gives information about command to be run when f.bin is a directory" do
+        allow(f).to receive(:keg_only?).and_return(true)
+        allow(f).to receive(:bin).and_return(Pathname.new("bin"))
+        allow(f.bin).to receive(:directory?).and_return(true)
+        expect(subject.caveats).to include("This formula is keg-only, which means it was not symlinked into #{HOMEBREW_PREFIX},\nbecause #{f.keg_only_reason.to_s.chomp}.\n\nIf you need to have this software first in your PATH run:\n  #{Utils::Shell.prepend_path_in_profile(f.opt_bin.to_s)}\n\n")
```

Comment:
Sorry for terrible previous message:

Sorry, no, I mean instead of checking for the long string being `include`d that you currently have just check for `include("formula is keg-only")`. This will mean that this test is more resistant (i.e. less likely to break) if we tweak the wording of this message in future. Does that make sense? Thanks!